internal/controller/services/monitoring/monitoring_controller_actions.go

```diff
@@ -74,3 +80,129 @@ func updatePrometheusConfigMap(ctx context.Context, rr *odhtypes.ReconciliationR
 		}
 	})
 }
+
+// handleInstrumentationCR manages OpenTelemetry Instrumentation CRs based on DSCI monitoring configuration
+func handleInstrumentationCR(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	dsci, err := cluster.GetDSCI(ctx, rr.Client)
+	if err != nil {
+		return fmt.Errorf("failed to get DSCInitialization instance: %w", err)
+	}
+
+	// Get the monitoring instance
+	monitoring, ok := rr.Instance.(*serviceApi.Monitoring)
+	if !ok {
+		return fmt.Errorf("instance is not of type *serviceApi.Monitoring")
+	}
+
+	instrumentationName := "opendatahub-instrumentation"
+	instrumentationNamespace := monitoring.Spec.Namespace
+
+	switch dsci.Spec.Monitoring.ManagementState {
```

Comment:
it seems that the Monitoring CR is never removed, hence I'd probably pass the ManagementState to the Monitoring spec if it is needed so there' won't be any need to lookup the DSCI.